// vite-plugin-md@0.1.5/dist/index.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }var __assign = Object.assign;

// src/index.ts
var _markdownit = require('markdown-it'); var _markdownit2 = _interopRequireDefault(_markdownit);
var _compilersfc = require('@vue/compiler-sfc');
var _graymatter = require('gray-matter'); var _graymatter2 = _interopRequireDefault(_graymatter);
function toArray(n) {
  if (!Array.isArray(n))
    return [n];
  return n;
}
function VitePluginMarkdown(options = {}) {
  const resolved = Object.assign({
    markdownItOptions: {},
    markdownItUses: [],
    markdownItSetup: () => {
    },
    wrapperClasses: "markdown-body",
    wrapperComponent: null,
    transforms: {}
  }, options);
  const markdown = new (0, _markdownit2.default)(__assign({
    html: true,
    linkify: true,
    typographer: true
  }, resolved.markdownItOptions));
  resolved.markdownItUses.forEach((e) => {
    const [plugin, options2] = toArray(e);
    markdown.use(plugin, options2);
  });
  resolved.markdownItSetup(markdown);
  const wrapperClasses = toArray(resolved.wrapperClasses).filter((i) => i).join(" ");
  return {
    transforms: [
      {
        test({path}) {
          return path.endsWith(".md");
        },
        transform(ctx) {
          const {isBuild, path} = ctx;
          let raw = ctx.code;
          if (resolved.transforms.before)
            raw = resolved.transforms.before(__assign(__assign({}, ctx), {code: raw}));
          const {content: md, data: frontmatter} = _graymatter2.default.call(void 0, raw);
          let sfc = markdown.render(md, {});
          if (resolved.wrapperClasses)
            sfc = `<div class="${wrapperClasses}">${sfc}</div>`;
          if (resolved.wrapperComponent)
            sfc = `<${resolved.wrapperComponent} :frontmatter="frontmatter">${sfc}</${resolved.wrapperComponent}>`;
          if (resolved.transforms.after)
            sfc = resolved.transforms.after(__assign(__assign({}, ctx), {code: sfc}));
          let {code: result} = _compilersfc.compileTemplate.call(void 0, {
            filename: path,
            id: path,
            source: sfc,
            transformAssetUrls: false
          });
          result = result.replace("export function render", "function render");
          result += `
const __matter = ${JSON.stringify(frontmatter)};`;
          result += "\nconst data = () => ({ frontmatter: __matter });";
          result += "\nconst __script = { render, data };";
          if (!isBuild)
            result += `
__script.__hmrId = ${JSON.stringify(path)};`;
          result += "\nexport default __script;";
          return result;
        }
      }
    ]
  };
}
var src_default = VitePluginMarkdown;


exports.default = src_default;
